fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/contract/routing/impl/SerializerRegistryImpl.java

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package org.hyperledger.fabric.contract.routing.impl;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.hyperledger.fabric.Logger;
+import org.hyperledger.fabric.contract.annotation.Serializer;
+import org.hyperledger.fabric.contract.execution.SerializerInterface;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ScanResult;
+
+/**
+ * Registry to hold permit access to the routing definitions. This is the
+ * primary internal data structure to permit access to information about the
+ * contracts, and their transaction functions.
+ *
+ * Contracts are added, and processed. At runtime, this can then be accessed to
+ * locate a specific 'Route' that can be handed off to the ExecutionService
+ *
+ * Type T should 
+ * 
+ */
+public class SerializerRegistryImpl {
+    private static Logger logger = Logger.getLogger(SerializerRegistryImpl.class);
+
+    private Class<Serializer> annotationClass;
```

Comment:
Why is this a field that presumably only gets set in the constructor, instead of a constant? (Not even sure a constant is necessary, or have I missed something?)